File: 01.Java-Basic/06.ConditionalStatementsAdvancedExercise/src/OnTimeForTheExam.java
import java.util.Scanner;

public class OnTimeForTheExam {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int examHours = Integer.parseInt(scanner.nextLine());
        int examMinutes = Integer.parseInt(scanner.nextLine());
        int myHours = Integer.parseInt(scanner.nextLine());
        int myMinutes = Integer.parseInt(scanner.nextLine());

        int totalExamMinutes = (examHours * 60) + examMinutes;
        int totalMyMinutes = (myHours * 60) + myMinutes;
        if (totalMyMinutes > totalExamMinutes) {
            System.out.println("Late");
            int diffInMinutes = totalMyMinutes - totalExamMinutes;
            if (diffInMinutes >= 60) {
                int diffHours = diffInMinutes / 60;
                int diffMinutes = diffInMinutes % 60;
                System.out.printf("%d:%02d hours after the start", diffHours, diffMinutes);
            }else {
                System.out.printf("%d minutes after the start", diffInMinutes);
            }
        } else {
            int diffInMinutes = totalExamMinutes - totalMyMinutes;
            if (diffInMinutes == 0) {
                System.out.println("On time");
            } else if (diffInMinutes <= 30) {
                System.out.println("On time");
                System.out.printf("%d minutes before the start", diffInMinutes);
            } else {
                System.out.println("Early");
                if (diffInMinutes >= 60) {
                    int diffHours = diffInMinutes / 60;
                    int diffMinutes = diffInMinutes % 60;
                    System.out.printf("%d:%02d hours before the start", diffHours, diffMinutes);
                } else {
                    System.out.printf("%d minutes before the start", diffInMinutes);
                }
            }
        }

    }
}

File: JavaScript/01.JavaScriptBasic/06Nested-Conditional-Statements-Еxercise/src/OnTimeForTheExam.js
function onTimeForTheExam(params) {
    let examHour = Number(params[0]);
    let examMinutes = Number(params[1]);
    let arrivalHour = Number(params[2]);
    let arrivalMinutes = Number(params[3]);
    let beforeOrAfter;
    let deltaMinutes = 0;
    let deltaHours = 0;
    let isOnTime;
 
    let examInMinutes = examHour * 60 + examMinutes;
    let arrivalInMinutes = arrivalHour * 60 + arrivalMinutes;
    deltaMinutes = examInMinutes - arrivalInMinutes;
 
    if (deltaMinutes < 0) {
        beforeOrAfter = "after";
        deltaMinutes *= -1;
        isOnTime = "Late";
    } else {
        beforeOrAfter = "before";
        if (deltaMinutes <= 30) {
            isOnTime = "On time";
        } else {
            isOnTime = "Early";
        }
    }
 
    console.log(isOnTime);
 
    if (deltaMinutes > 0 && deltaMinutes <= 59) {
        console.log(`${deltaMinutes} minutes ${beforeOrAfter} the start`)
    } else if (deltaMinutes > 59) {
        deltaHours = Math.floor(deltaMinutes / 60);
        deltaMinutes = deltaMinutes % 60;
        if (deltaMinutes < 10) {
            console.log(`${deltaHours}:0${deltaMinutes} hours ${beforeOrAfter} the start`);
        } else {
            console.log(`${deltaHours}:${deltaMinutes} hours ${beforeOrAfter} the start`);
        }
    }
}